### PR TITLE
Allowed to upload images in WebP format

### PR DESCRIPTION
**Pull request by empiricompany**
### Description (*)
Since the webp format is now supported by all modern browsers *, I have added the ability to upload images in webp format.
* https://caniuse.com/webp

This PR allow to upload an image in webp format and add support to resize and cache it correctly in frontend.

PS: I know that there are extensions/techniques to dynamically generate the WebP version on the frontend, but based on the target audience, with this PR, we can decide to upload them directly in the correct format.

### Manual testing scenarios (*)
Test all scenarios with an image in webp format

Tested scenarios:
- [x] Product images
- [x] Product Custom Options Input Type: File
immagine
- [x] Category Image and Thumbnail Image
- [x] CMS WYSIWYG Media Storage

Untested scenarios:
- [ ] Import/export
- [ ] API

I create this pull request as a draft since there are areas that I have not yet tested.
Tested on PHP 7.4.29 with GD 2.1.0 (support to webp should be enabled by default)

### CODE REVIEW

**Comment by empiricompany** [comment]
@fballiano can you rebase this PR since this commit?
[first draft webp support catalog media product and cms](https://github.com/OpenMage/magento-lts/pull/3384/commits/c6ceb853e42df038766625cbe36d7c054d591fa5)

Sorry if I always make a huge mess. 😅 but what matters is the substance

File: app/code/core/Mage/Eav/Model/Attribute/Data/Image.php [hunk 1]
```diff
@@ -47,6 +47,7 @@ protected function _validateByRules($value)
             1   => 'gif',
             2   => 'jpg',
             3   => 'png',
+            18  => 'webp'
```
Comment (fballiano) [inline app/code/core/Mage/Eav/Model/Attribute/Data/Image.php:50]: just out of curiosity, why is that 18?

**Comment by fballiano** [comment]
the PR looks ok, I don't think there's anything to rebase :-)

**Comment by fballiano** [comment]
but actually I'd like to see https://github.com/OpenMage/magento-lts/pull/3301 merged before this one, since they work on the same stuff

File: app/code/core/Mage/Eav/Model/Attribute/Data/Image.php (hunk 1 above)
Comment (empiricompany) [reply app/code/core/Mage/Eav/Model/Attribute/Data/Image.php:50]: it's IMAGETYPE_WEBP constant

Comment (fballiano) [reply app/code/core/Mage/Eav/Model/Attribute/Data/Image.php:50]: ahhh ok ok thanks!

**Comment by fballiano** [comment]
:-) actually we could merge #3301 so this PR now needs a bit of reworking, but it's a good thing

**Comment by fballiano** [comment]
I think I fixed the conflicts, hope I didn't screw anything in the process

**Comment by fballiano** [comment]
IMHO it would be great to have a single 

```
    /**
     * @var string[]
     */
    public const ALLOWED_IMAGES_EXTENSIONS = ['webp', 'jpg', 'jpeg', 'png', 'gif', 'bmp'];
```

instead of many arrays throughout the code, all of the code should be referencing to that constant :-)

**Comment by empiricompany** [comment]
> IMHO it would be great to have a single
> 
> ```
>     /**
>      * @var string[]
>      */
>     public const ALLOWED_IMAGES_EXTENSIONS = ['webp', 'jpg', 'jpeg', 'png', 'gif', 'bmp'];
> ```
> 
> instead of many arrays throughout the code, all of the code should be referencing to that constant :-)

yes but it was keep untouched for backward compatibility reasons
maybe we can create new constant and return it keeping in the deprecated methods?

**Comment by fballiano** [comment]
the one that I posted is a new one ;-)

**Comment by empiricompany** [comment]
> Personally, I would not put webp as the first image format in the list that is seen in the Backend. I would list it at the end, because it is a format after all the ones that have been so far.

sorry i can't figure out where is showed

**Comment by fballiano** [comment]
he says only in the array list

**Comment by empiricompany** [comment]
> Personally, I would not put webp as the first image format in the list that is seen in the Backend. I would list it at the end, because it is a format after all the ones that have been so far.

done

**Comment by fballiano** [comment]
I've unified all into a single constant, see if you like it :-)

**Comment by ADDISON74** [comment]
So far so good. 

I would like to test "**Image type product options**" mentioned in the description, but I don't know anything about this feature. Any help?

**Comment by fballiano** [comment]
not sure but probably you have to create a new product attribute of type image:

Screenshot 2023-07-18 alle 21 26 14

**Comment by ADDISON74** [comment]
That type of attribute adds a new column in the Images section. I asked because he made me curious about what it could be and I don't already know.

screenshot

**Comment by fballiano** [comment]
ahhhh ok then I don't know 😅

**Comment by empiricompany** [comment]
@ADDISON74 you have to create a product custom options of type image that allow customers to upload an image in the frontend

**Comment by ADDISON74** [comment]
Now I understand what it is about. It should have been mentioned as "File type in custom options" . A word led to misunderstanding.

**Review by ADDISON74 — APPROVED** [review]
I tested this PR and encountered no issues. It is a new feature in OpenMage much discussed in the forums of the time.

**Comment by empiricompany** [comment]
> Now I understand what it is about. It should have been mentioned as "File type in custom options" . A word led to misunderstanding.

yes sorry i've updated PR description

**Comment by fballiano** [comment]
great, let's merge it, it's a nice little new addition for the upcoming release :-)

**Comment by ADDISON74** [comment]
**[Only if it is confirmed]**

I would add a warning in the README file that issues may arise with Magmi. Here is a report https://github.com/dweeves/magmi-git/issues/596.

**Comment by empiricompany** [comment]
> I would add a warning in the README file that issues may arise with Magmi. Here is a report [dweeves/magmi-git#596](https://github.com/dweeves/magmi-git/issues/596).

Maybe now it works, we need to try

**Comment by ADDISON74** [comment]
First we have to do a test for the Import feature we already have in OpenMage and confirm that it works (CSV file, a few images uploaded, ...). I will check these days.

**Pull request merged**